docs/changelog/4_x.rst

```diff
@@ -0,0 +1,467 @@
+====
+v4.0
+====
+
+:edb-alt-title: EdgeDB v4
+
+To play with the new features, install the CLI using `our installation guide
+<https://www.edgedb.com/install>`_ and initialize a new project.
+
+.. code-block:: bash
+
+  $ edgedb project init --server-version 4.0-beta.1
+
+
+Upgrading
+=========
+
+**Local instances**
+
+To upgrade a local project, first ensure that your CLI is up to date with
+``edgedb cli upgrade``. Then run the following command inside the project
+directory.
+
+.. code-block:: bash
+
+  $ edgedb project upgrade --to-testing
+
+Alternatively, specify an instance name if you aren't using a project.
+
+.. code-block:: bash
+
+  $ edgedb instance upgrade --to-testing -I my_instance
+
+**Hosted instances**
```

Comment:
Should mention here that v4.0 only supports PostgreSQL >= 14 (dropping PostgreSQL 13), so users might need to upgrade Postgres before upgrading EdgeDB.